webapp/src/main/java/com/box/l10n/mojito/service/tm/textunitdtocache/TextUnitDTOsSmileCacheBlobStorage.java

```diff
@@ -0,0 +1,53 @@
+package com.box.l10n.mojito.service.tm.textunitdtocache;
+
+import static com.box.l10n.mojito.service.blobstorage.StructuredBlobStorage.Prefix.TEXT_UNIT_DTOS_CACHE;
+
+import com.box.l10n.mojito.json.ObjectMapper;
+import com.box.l10n.mojito.service.blobstorage.Retention;
+import com.box.l10n.mojito.service.tm.search.TextUnitDTO;
+import com.google.common.collect.ImmutableList;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+@Component
+@ConditionalOnProperty(name = "l10n.cache.textunit.smile.enabled", havingValue = "true")
+public class TextUnitDTOsSmileCacheBlobStorage extends TextUnitDTOsCacheBlobStorage {
```

Comment:
can we just switch the object mapper based on the configuration instead of using class extension? basically just have a getObject mapper with a if statement?

Reply:
Sure I can update it to be contained within a single class

Reply:
We could but as the smile object mapper retrieves & writes bytes instead of strings it would mean a lot of methods in the same class that achieve the same function by slightly different means, I felt it was cleaner to separate it into a separate class rather than have a lot of if statements throughout the existing class. 

We can't just replace the `objectMapper` calls in the class with a `getObjectMapper()` function as depending on the configuration we'll be writing or retrieving via different `structuredBlobStorage` methods.